tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,236 @@
+        parser.add_argument("-y", "--fiscal_year", type=str, help="Re-parse all files in the fiscal year, e.g. 2021.")
+        parser.add_argument("-a", "--all", action='store_true', help="Clean and re-parse all datafiles. If selected, "
+                            "fiscal_year/quarter aren't necessary.")
+        parser.add_argument("-n", "--new_indices", action='store_true', help="Move re-parsed data to new Elastic "
+                            "indices.")
+        parser.add_argument("-d", "--delete_indices", action='store_true', help="Requires new_indices. Delete the "
+                            "current Elastic indices.")
+
+    def __get_log_context(self, system_user):
+        """Return logger context."""
+        context = {'user_id': system_user.id,
+                   'action_flag': ADDITION,
+                   'object_repr': "Clean and Re-parse"
+                   }
+        return context
+
+    def __backup(self, backup_file_name, log_context):
+        """Execute Postgres DB backup."""
+        try:
+            logger.info("Beginning re-parse DB Backup.")
+            call_command('backup_db', '-b', '-f', f'{backup_file_name}')
+            logger.info("Backup complete! Commencing clean and re-parse.")
+
+            log("Database backup complete.",
+                logger_context=log_context,
+                level='info')
+        except Exception as e:
+            log("Database backup FAILED. Clean and re-parse NOT executed. Database and Elastic are CONSISTENT!",
+                logger_context=log_context,
+                level='error')
+            raise e
+
+    def __handle_elastic(self, new_indices, delete_indices, log_context):
+        """Create new Elastic indices and delete old ones."""
+        if new_indices:
+            try:
+                if not delete_indices:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias', '--use-alias-keep-index')
+                else:
+                    call_command('tdp_search_index', '--create', '-f', '--use-alias')
+                log("Index creation complete.",
+                    logger_context=log_context,
+                    level='info')
+            except Exception as e:
+                log("Elastic index creation FAILED. Clean and re-parse NOT executed. "
+                    "Database is CONSISTENT, Elastic is INCONSISTENT!",
+                    logger_context=log_context,
+                    level='error')
+                raise e
+            # TODO: Need to ask Alex if we can run some queries in prod to deduce the average number of records per DF.
+
+    def __delete_records(self, docs, file_ids, new_indices, log_context):
+        """Delete records and documents from Postgres and Elastic respectively."""
+        total_deleted = 0
+        for doc in docs:
+            # atomic delete?
+            try:
+                model = doc.Django.model
+                qset = model.objects.all().filter(datafile_id__in=file_ids)
```

Comment:
I guess you can move the query to outside the loop?!!

Reply:
Could you explain how I might be able to move the query out of the loop?